WordPress/Classes/ViewRelated/Reader/ReaderPostCardCell.swift

```diff
@@ -278,51 +298,94 @@ protocol ReaderTopicsChipsDelegate: class {
+        })
+
+        mediaRequestAuthenticator.authenticatedRequest(
+            for: url,
+            from: host,
+            onComplete: { request in
+                self.avatarImageView.downloadImage(usingRequest: request)
+                self.avatarImageView.isHidden = false
             },
-                onFailure: { error in
-                    CrashLogging.logError(error)
-                    self.avatarImageView.isHidden = true
+            onFailure: { error in
+                CrashLogging.logError(error)
+                self.avatarImageView.isHidden = true
             })
-        } else {
-            avatarImageView.isHidden = true
+    }
+
+    private func setAuthorAvatar() {
+        guard isWPForTeams,
+              let contentProvider = contentProvider,
+              let url = contentProvider.avatarURLForDisplay() else {
+            authorAvatarImageView.isHidden = true
+            return
+        }
+
+        authorAvatarImageView.isHidden = false
+        authorAvatarImageView.downloadImage(from: url, placeholderImage: Constants.authorAvatarPlaceholderImage)
+    }
+
+    private func setBlogLabels() {
+        guard let contentProvider = contentProvider else {
+            return
+        }
+
+        authorNameLabel.isHidden = !isWPForTeams
+        arrowImageView.isHidden = !isWPForTeams
+
+        if isWPForTeams {
+            authorNameLabel.text = contentProvider.authorForDisplay()
+            configureArrowImage()
         }
 
-        blogNameLabel.text = contentProvider.blogNameForDisplay() ?? ""
-        blogHostNameLabel.text = contentProvider.siteHostNameForDisplay() ?? ""
+        blogNameLabel.text = contentProvider.blogNameForDisplay()
+        blogHostNameLabel.text = contentProvider.siteHostNameForDisplay()
 
         let dateString: String = datePublished()
         bylineSeparatorLabel.isHidden = dateString.isEmpty
         bylineLabel.text = dateString
     }
 
-    fileprivate func configureAvatarImageView() {
-        avatarImageView.layer.borderColor = WPStyleGuide.readerCardBlogIconBorderColor().cgColor
-        avatarImageView.layer.borderWidth = Constants.imageBorderWidth
-        avatarImageView.layer.masksToBounds = true
+    private func configureArrowImage(withTint tint: UIColor = WPStyleGuide.readerCardBlogNameLabelTextColor()) {
+        arrowImageView.image = UIImage.gridicon(.dropdown).imageWithTintColor(tint)
+
+        let imageRotationAngle = (userInterfaceLayoutDirection() == .rightToLeft) ?
+            Constants.rotate90Degrees :
+            Constants.rotate270Degrees
+
+        arrowImageView.transform = CGAffineTransform(rotationAngle: imageRotationAngle)
+    }
+
+    private func configureAvatarImageView(_ imageView: UIImageView) {
+        imageView.layer.borderColor = WPStyleGuide.readerCardBlogIconBorderColor().cgColor
+        imageView.layer.borderWidth = Constants.imageBorderWidth
+        imageView.layer.masksToBounds = true
     }
```

Comment:
The ReaderCrossPostCell avatars could benefit from this treatment too. wdyt?

![Simulator Screen Shot - iPhone 11 Pro - 2020-11-17 at 14 32 29](https://user-images.githubusercontent.com/1062444/99451141-5f846280-28e7-11eb-8456-830fd88ba3e5.png)

Reply:
Sure. I'll look at it separately though if that's ok.